src/main/java/com/firebolt/jdbc/service/FireboltEngineService.java

```diff
@@ -1,117 +1,93 @@
-	private static final String ERROR_NO_ENGINE_ATTACHED = "There is no Firebolt engine running on %s attached to the database %s. To connect first make sure there is a running engine and then try again.";
-	private static final String ERROR_NO_ENGINE_WITH_NAME = "There is no Firebolt engine running on %s with the name %s. To connect first make sure there is a running engine and then try again.";
-	private final FireboltAccountClient fireboltAccountClient;
-
-	/**
-	 * Returns the engine
-	 *
-	 * @param connectionUrl   the connection url
-	 * @param loginProperties properties to login
-	 * @param accessToken     the access token
-	 * @return the engine
-	 */
-	public Engine getEngine(String connectionUrl, FireboltProperties loginProperties, String accessToken)
-			throws FireboltException {
-		String accountId = null;
-		Engine engine;
-		try {
-			if (StringUtils.isNotEmpty(loginProperties.getAccount())) {
-				accountId = getAccountId(connectionUrl, loginProperties.getAccount(), accessToken).orElse(null);
-			}
-			if (StringUtils.isEmpty(loginProperties.getEngine())) {
-				engine = getDefaultEngine(connectionUrl, accountId, loginProperties.getDatabase(), accessToken);
-			} else {
-				engine = getEngineWithName(connectionUrl, accountId, loginProperties.getEngine(), accessToken);
-			}
-		} catch (FireboltException e) {
-			throw e;
-		} catch (Exception e) {
-			throw new FireboltException("Failed to get engine", e);
-		}
-		validateEngineIsNotStarting(engine);
-		return engine;
-	}
-
-	private Engine getEngineWithName(String connectionUrl, String accountId, String engineName, String accessToken)
-			throws FireboltException, IOException {
-		FireboltEngineIdResponse response = fireboltAccountClient.getEngineId(connectionUrl, accountId, engineName,
-				accessToken);
-		String engineID = Optional.ofNullable(response).map(FireboltEngineIdResponse::getEngine)
-				.map(FireboltEngineIdResponse.Engine::getEngineId).orElseThrow(() -> new FireboltException(
-						"Failed to extract engine id field from the server response: the response from the server is invalid."));
-		FireboltEngineResponse fireboltEngineResponse = fireboltAccountClient.getEngine(connectionUrl, accountId,
-				engineName, engineID, accessToken);
+    private static final String ENGINE_URL = "url";
+    private static final String ENGINE_NAME = "engine_name";
+    private static final String STATUS_FIELD_NAME = "status";
+    private static final String DEFAULT_ENGINE_QUERY = "SELECT engs.url, engs.status, engs.engine_name\n" +
+            "FROM information_schema.databases AS dbs\n" +
+            "INNER JOIN information_schema.engines AS engs\n" +
+            "ON engs.attached_to = dbs.database_name\n" +
+            "AND engs.engine_name = NULLIF(SPLIT_PART(ARRAY_FIRST(\n" +
+            "        eng_name -> eng_name LIKE '%%(default)',\n" +
+            "        SPLIT(',', attached_engines)\n" +
+            "    ), ' ', 1), '')\n" +
+            "WHERE database_name = ?";
```

Comment:
I believe the concept of a default engine will go away soon. The logic agreed at the moment is if no engine name is specified by the user - connect to a system engine.